Client/Frontend/Home/Wallpapers/v2/UI/WallpaperSelectorViewController.swift

```diff
@@ -0,0 +1,251 @@
+                                                                                size: 16)
+        button.titleLabel?.textAlignment = .center
+        button.setTitle(.Onboarding.WallpaperSelectorAction, for: .normal)
+        button.accessibilityIdentifier = AccessibilityIdentifiers.Onboarding.Wallpaper.settingsButton
+    }
+
+    // MARK: - Initializers
+    init(viewModel: WallpaperSelectorViewModel,
+         notificationCenter: NotificationProtocol = NotificationCenter.default) {
+        self.viewModel = viewModel
+        self.notificationCenter = notificationCenter
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View setup & lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        applyTheme()
+        setupNotifications(forObserver: self, observing: [.DisplayThemeChanged])
+        settingsButton.addTarget(self, action: #selector(self.settingsButtonTapped), for: .touchUpInside)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        applyTheme()
+
+        // make collection view fixed height so the bottom sheet can size correctly
+        let height = collectionView.collectionViewLayout.collectionViewContentSize.height +
+            WallpaperSelectorViewController.UX.cardShadowHeight
+        collectionViewHeightConstraint.constant = height
+        view.layoutIfNeeded()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass
+            || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            updateOnRotation()
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        updateOnRotation()
+    }
+}
+
+// MARK: - CollectionView Data Source
+
+extension WallpaperSelectorViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return viewModel.wallpaperCellModels.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: WallpaperCollectionViewCell.cellIdentifier,
+                                                            for: indexPath) as? WallpaperCollectionViewCell,
+                let cellViewModel = viewModel.wallpaperCellModels[safe: indexPath.row] else {
+            return UICollectionViewCell()
+        }
```

Comment:
```suggestion
        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: WallpaperCollectionViewCell.cellIdentifier,
                                                            for: indexPath) as? WallpaperCollectionViewCell,
                let cellViewModel = viewModel.wallpaperCellModels[safe: indexPath.row] 
        else { return UICollectionViewCell() }
```
nit: guard/else on separate line. Also, identation. I think.